analytics_data_api/v0/views/learners.py

```diff
@@ -0,0 +1,113 @@
+"""
+API methods for module level data.
+"""
+from django.http import Http404
+import re
+from rest_framework.exceptions import ParseError
+from rest_framework import generics
+
+from analytics_data_api.v0.models import RosterEntry
+from analytics_data_api.v0.serializers import LearnerSerializer
+
+
+class CourseViewMixin(object):
+    """
+    Captures the course_id query arg and validates it.
+    """
+
+    COURSE_ID_PATTERN = r'(?P<course_id>[^/+]+[/+][^/+]+[/+][^/]+)'
```

Comment:
This is also defined in `analytics_data_api/v0/urls/courses.py`. Could we share it?